app/controllers/admin/layouts_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Admin::LayoutsController < ApplicationController
+  before_action :load_grouping, only: %i(edit update)
+
+  def edit
+    @grouping_name = @grouping.name
+    @desks = @grouping.desks
+    @child_groupings = @grouping.child_groupings
+  end
+
+  def update
+    lay_out = JSON.parse(params.require(:layout))
+    @grouping.children.each do |area|
+      x, y = lay_out[area.class.name.downcase][area.id.to_s].values_at('x', 'y')
+      if area.coordinate.nil?
```

Comment:
Nit: Generally, for readability, the positive case comes before the negative as in:
```ruby
if area.coordinate
  ...
else
  ...
```